### DISCUSSION TITLE
How do I cutomize tooltip like on KLineChartPro?

### DISCUSSION THREAD

**Discussion by rxdiscovery** [klinecharts/KLineChart#563] [Q&A] [answered]
Hello,

I'd like to know how to customize tooltip to add icons with events, like on KLineChartPro ? I tried to see the project code, but I didn't understand the logic behind it.

![tooltips_klinechart](https://github.com/klinecharts/KLineChart/assets/131091294/2091b3ec-dcb1-45e0-a302-c176cb6d7419)

I'm really stuck on this point :/ 

Thank you in advance for your help..

**Comment by char101** [comment, marked as answer]
It is a unicode character. Using `chart.setStyles` you set the character used for the eye and config icon.

See https://github.com/klinecharts/KLineChart/blob/045973502f210bd92c060c0ac0cb136d64777106/src/common/Styles.ts#L177

Example
```javascript
chart.setStyles({
  indicator: {
    tooltip: {
      icons: [
        { id: 'l', position: 'left', icon: 'L', color: '#FFF' },
        { id: 'm', position: 'middle', icon: 'M' },
        { id: 'r', position: 'right', icon: 'R' },
      ],
    }
  }
});
```

**Reply by ct152098** [reply]
10版本的怎么加？
